luigi/contrib/kubernetes.py

```diff
@@ -181,27 +203,90 @@ def signal_complete(self):
         """
         pass
 
+    def __get_pods(self):
+        pod_objs = Pod.objects(self.__kube_api) \
+            .filter(selector="job-name=" + self.uu_name) \
+            .response['items']
+        return [Pod(self.__kube_api, p) for p in pod_objs]
+
+    def __get_job(self):
+        jobs = Job.objects(self.__kube_api) \
+            .filter(selector="luigi_task_id=" + self.job_uuid) \
+            .response['items']
+        assert len(jobs) == 1, "Kubernetes job " + self.uu_name + " not found"
+        return Job(self.__kube_api, jobs[0])
+
+    def __print_pod_logs(self):
+        for pod in self.__get_pods():
+            logs = pod.logs(timestamps=True).strip()
+            self.__logger.info("Fetching logs from " + pod.name)
+            if len(logs) > 0:
+                for l in logs.split('\n'):
+                    self.__logger.info(l)
+
+    def __print_kubectl_hints(self):
+        self.__logger.info("To stream Pod logs, use:")
+        for pod in self.__get_pods():
+            self.__logger.info("`kubectl logs -f pod/%s`" % pod.name)
+
+    def __verify_job_has_started(self):
+        """Asserts that the job has successfully started"""
+        # Verify that the job started
+        self.__get_job()
+
+        # Verify that the pod started
+        pods = self.__get_pods()
+
+        assert len(pods) > 0, "No pod scheduled by " + self.uu_name
+        for pod in pods:
+            status = pod.obj['status']
+            for cont_stats in status.get('containerStatuses', []):
+                if 'terminated' in cont_stats['state']:
+                    t = cont_stats['state']['terminated']
+                    err_msg = "Pod %s %s (exit code %d). Logs: `kubectl logs pod/%s`" % (
+                        pod.name, t['reason'], t['exitCode'], pod.name)
+                    assert t['exitCode'] == 0, err_msg
+
+                assert 'waiting' not in cont_stats['state'],\
```

Comment:
Does this mean that a waiting pod is going to fail? Sometimes we want to have more pods than workers, so we can have a sort of queuing system.